src/inmanta/util.py

```diff
@@ -699,3 +701,24 @@ async def __aenter__(self) -> T:
 
     async def __aexit__(self, *excinfo: object) -> None:
         pass
+
+
+async def join_threadpools(threadpools: List[ThreadPoolExecutor]) -> None:
+    # idea borrowed from BaseEventLoop.shutdown_default_executor
```

Comment:
This comment is good but I think we also need a comment as to why we implement a custom method for this rather than to rely on plain `threadpool.shutdown`.